automation/build-bin.ts

```diff
@@ -77,14 +149,101 @@ export async function buildPkg() {
 	);
 }
 
+/**
+ * Create the zip file for the standalone 'pkg' bundle previously created
+ * by the buildPkg() function in 'build-bin.ts'.
+ */
+async function zipPkg() {
+	const outputFile = standaloneZips[process.platform];
+	if (!outputFile) {
+		throw new Error(
+			`Standalone installer unavailable for platform "${process.platform}"`,
+		);
+	}
+	await fs.mkdirp(path.dirname(outputFile));
+	await new Bluebird((resolve, reject) => {
+		console.log(`Zipping standalone package to "${outputFile}"...`);
+
+		const archive = archiver('zip', {
+			zlib: { level: 7 },
+		});
+		archive.directory(path.join(ROOT, 'build-bin'), 'balena-cli');
+
+		const outputStream = fs.createWriteStream(outputFile);
+
+		outputStream.on('close', resolve);
+		outputStream.on('error', reject);
+
+		archive.on('error', reject);
+		archive.on('warning', console.warn);
+
+		archive.pipe(outputStream);
+		archive.finalize();
+	});
+}
+
+export async function buildStandaloneZip() {
+	console.log(`Building standalone zip package for CLI ${version}`);
+	try {
+		await buildPkg();
+		await zipPkg();
+	} catch (error) {
+		console.log(`Error creating standalone zip package: ${error}`);
+		process.exit(1);
+	}
+	console.log(`Standalone zip package build completed`);
+}
+
+async function renameInstallerFiles() {
+	if (await fs.pathExists(oclifInstallers[process.platform])) {
+		await fs.rename(
+			oclifInstallers[process.platform],
+			renamedOclifInstallers[process.platform],
+		);
+	}
+}
+
+/**
+ * If the CSC_LINK and CSC_KEY_PASSWORD env vars are set, digitally sign the
+ * executable installer by running the balena-io/scripts/shared/sign-exe.sh
+ * script (which must be in the PATH) using a MSYS2 bash shell.
+ */
+async function signWindowsInstaller() {
+	if (process.env.CSC_LINK && process.env.CSC_KEY_PASSWORD) {
+		const exeName = renamedOclifInstallers[process.platform];
+		console.log(`Signing installer "${exeName}"`);
+		await runUnderMsys([
+			'sign-exe.sh',
+			'-f',
+			exeName,
+			'-d',
+			`balena-cli ${version}`,
+		]);
+	} else {
+		console.log(
+			'Skipping installer signing step because CSC_* env vars are not set',
+		);
+	}
+}
+
```

Comment:
We already discussed to add a comment here explaining the differences between what oclif does in osx vs win. Just adding this as a reminder

Reply:
Good that you raised this, because on review I see that setting the OSX_KEYCHAIN env var at that point was useless (at that point, `pkgbuild` had already run). I've deleted that code and added a comment to explain that the Windows installer is explicitly signed by a script invoked by balena-cli, whereas the macOS installer is signed by oclif which runs the `pkbuild` utility.